[bug]: Multi-tenant PgVector startup fails: CREATE EXTENSION has no schema to create in (SQLSTATE 3F000)

## Summary

On startup, `MultiTenantPgVectorLoader` crashes while initializing the PgVector schema for a tenant (e.g. `000001`):

```
Caused by: java.lang.RuntimeException: Unable to execute run with tenant ID 000001
    at com.bytechef.tenant.TenantContext.runWithTenantId(TenantContext.java:91)
    at com.bytechef.ee.tenant.multi.pgvector.MultiTenantPgVectorLoader.initializeSchema(MultiTenantPgVectorLoader.java:70)
    ...
Caused by: org.springframework.jdbc.UncategorizedSQLException: StatementCallback;
    uncategorized SQLException for SQL [CREATE EXTENSION IF NOT EXISTS vector];
    SQL state [3F000]; ERROR: no schema has been selected to create in
```

## Root cause

`MultiTenantPgVectorDataSource` (via `BaseDataSource.setSearchPath`) pins `search_path` to the tenant's vector schema — e.g. `bytechef_vectorstore_000001` — on every connection it hands out.

`MultiTenantPgVectorLoader.doInitializeSchema` runs `CREATE EXTENSION IF NOT EXISTS vector` **before** `CREATE SCHEMA IF NOT EXISTS <schemaName>`. Because `CREATE EXTENSION` with no `SCHEMA` clause targets the first schema on the search_path, and that tenant schema doesn't exist yet, PostgreSQL aborts with `3F000: no schema has been selected to create in`.

Note: simply reordering (schema first) is **not** sufficient/correct — PostgreSQL extensions are database-global objects assigned to exactly one schema, and all tenants share a single database (the datasource only flips `search_path`, there is no database-per-tenant). The extension must live in a single shared schema. The table DDL already references the type as `public.vector(...)`, so `public` is the intended home.

## Fix

Install the extensions into `public` explicitly so they are independent of the tenant `search_path` and shared across tenants:

```java
jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS vector SCHEMA public");
jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS hstore SCHEMA public");
// uuid-ossp SCHEMA public when idType == UUID
```

`CREATE EXTENSION IF NOT EXISTS ... SCHEMA public` is idempotent across every tenant's init. The remaining statements (`CREATE SCHEMA`, `CREATE TABLE`, `CREATE INDEX`) are already fully schema-qualified and do not depend on search_path.

## Affected file

- `server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/pgvector/MultiTenantPgVectorLoader.java`

## Follow-up

Add a Testcontainers `IntTest` (real Postgres + pgvector) that boots the loader with the tenant `search_path` pinned to a not-yet-created schema and asserts init succeeds — guards against regression. The existing `MultiTenantPgVectorStoreTest` uses a mocked `JdbcTemplate` and cannot reproduce the server-side 3F000 error.

--- a/.gitguardian.yaml
+++ b/.gitguardian.yaml
@@ -1,0 +1,8 @@
+version: 2
+secret:
+  ignored-matches:
+    # Throwaway password for the ephemeral FTP/SFTP Testcontainers used by
+    # BaseFtpActionIntTest. It only authenticates against a container that is
+    # created and destroyed within the test run — nothing to rotate or revoke.
+    - name: FTP Testcontainers fixture password (false positive)
+      match: byteChefFtpPass123

--- a/server/apps/server-app/src/main/resources/config/application-prod.yml
+++ b/server/apps/server-app/src/main/resources/config/application-prod.yml
@@ -44,4 +44,6 @@ bytechef:
 #  help-hub:
 #    enabled: true
   public-url: http://127.0.0.1:8080
+  user-guiding:
+    enabled: true
   webhook-url: ${bytechef.public-url}/webhooks/{id}

--- a/server/ee/libs/config/tenant-multi-pgvector-config/build.gradle.kts
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/build.gradle.kts
@@ -12,8 +12,14 @@ dependencies {
     implementation(project(":server:libs:core:tenant:tenant-api"))
     implementation(project(":server:libs:platform:platform-api"))
 
+    testImplementation("com.zaxxer:HikariCP")
     testImplementation("org.assertj:assertj-core")
     testImplementation("org.junit.jupiter:junit-jupiter")
     testImplementation("org.mockito:mockito-core")
     testImplementation("org.mockito:mockito-junit-jupiter")
+    testImplementation("org.testcontainers:testcontainers")
+    testImplementation("org.testcontainers:postgresql")
+    testImplementation("org.testcontainers:junit-jupiter")
+
+    testRuntimeOnly("org.postgresql:postgresql")
 }

--- a/server/ee/libs/config/tenant-multi-pgvector-config/build.gradle.kts
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/build.gradle.kts
@@ -12,7 +12,6 @@ dependencies {
     implementation(project(":server:libs:core:tenant:tenant-api"))
     implementation(project(":server:libs:platform:platform-api"))
 
-    testImplementation("com.zaxxer:HikariCP")
     testImplementation("org.assertj:assertj-core")
     testImplementation("org.junit.jupiter:junit-jupiter")
     testImplementation("org.mockito:mockito-core")

--- a/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/pgvector/MultiTenantPgVectorLoader.java
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/src/main/java/com/bytechef/ee/tenant/multi/pgvector/MultiTenantPgVectorLoader.java
@@ -79,11 +79,11 @@ public class MultiTenantPgVectorLoader implements InitializingBean {
 
         log.info("Initializing PgVectorStore schema for table: {} in schema: {}", tableName, schemaName);
 
-        jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS vector");
-        jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS hstore");
+        jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS vector SCHEMA public");
+        jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS hstore SCHEMA public");
 
         if (properties.getIdType() == PgVectorStore.PgIdType.UUID) {
-            jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\"");
+            jdbcTemplate.execute("CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\" SCHEMA public");
         }
 
         jdbcTemplate.execute("CREATE SCHEMA IF NOT EXISTS " + schemaName);

--- a/server/ee/libs/config/tenant-multi-pgvector-config/src/test/java/com/bytechef/ee/tenant/multi/pgvector/MultiTenantPgVectorLoaderIntTest.java
+++ b/server/ee/libs/config/tenant-multi-pgvector-config/src/test/java/com/bytechef/ee/tenant/multi/pgvector/MultiTenantPgVectorLoaderIntTest.java
@@ -1,0 +1,113 @@
+/*
+ * Copyright 2025 ByteChef
+ *
+ * Licensed under the ByteChef Enterprise license (the "Enterprise License");
+ * you may not use this file except in compliance with the Enterprise License.
+ */
+
+package com.bytechef.ee.tenant.multi.pgvector;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.bytechef.ee.tenant.multi.sql.MultiTenantPgVectorDataSource;
+import com.bytechef.tenant.TenantContext;
+import com.bytechef.tenant.service.TenantService;
+import com.zaxxer.hikari.HikariDataSource;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.ai.vectorstore.pgvector.PgVectorStore;
+import org.springframework.ai.vectorstore.pgvector.autoconfigure.PgVectorStoreProperties;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * @version ee
+ *
+ * @author Ivica Cardic
+ */
+@Testcontainers
+class MultiTenantPgVectorLoaderIntTest {
+
+    private static final String TABLE_NAME = "kb_vector_store";
+    private static final String TENANT_ID = "000001";
+    private static final String VECTOR_SCHEMA = "bytechef_vectorstore_" + TENANT_ID;
+
+    @Container
+    private static final PostgreSQLContainer<?> postgreSQLContainer = new PostgreSQLContainer<>(
+        DockerImageName.parse("pgvector/pgvector:pg16")
+            .asCompatibleSubstituteFor("postgres"));
+
+    private HikariDataSource hikariDataSource;
+    private JdbcTemplate plainJdbcTemplate;
+    private JdbcTemplate tenantJdbcTemplate;
+
+    @BeforeEach
+    void setUp() {
+        hikariDataSource = new HikariDataSource();
+
+        hikariDataSource.setJdbcUrl(postgreSQLContainer.getJdbcUrl());
+        hikariDataSource.setUsername(postgreSQLContainer.getUsername());
+        hikariDataSource.setPassword(postgreSQLContainer.getPassword());
+
+        // Tenant-aware template: each borrowed connection runs SET search_path TO bytechef_vectorstore_000001,
+        // which does not exist until the loader creates it. This is the production wiring that triggered 3F000.
+        tenantJdbcTemplate = new JdbcTemplate(new MultiTenantPgVectorDataSource(hikariDataSource));
+
+        // Plain template (default search_path) used to assert the resulting catalog state.
+        plainJdbcTemplate = new JdbcTemplate(hikariDataSource);
+    }
+
+    @AfterEach
+    void tearDown() {
+        TenantContext.resetCurrentTenantId();
+
+        if (hikariDataSource != null) {
+            hikariDataSource.close();
+        }
+    }
+
+    @Test
+    void testAfterPropertiesSetInstallsExtensionInPublicWhenTenantSchemaMissing() {
+        TenantService tenantService = mock(TenantService.class);
+
+        when(tenantService.getTenantIds()).thenReturn(List.of(TENANT_ID));
+
+        PgVectorStoreProperties properties = new PgVectorStoreProperties();
+
+        properties.setDimensions(1536);
+        properties.setIdType(PgVectorStore.PgIdType.UUID);
+        properties.setIndexType(PgVectorStore.PgIndexType.HNSW);
+        properties.setDistanceType(PgVectorStore.PgDistanceType.COSINE_DISTANCE);
+
+        MultiTenantPgVectorLoader loader = new MultiTenantPgVectorLoader(
+            tenantJdbcTemplate, properties, TABLE_NAME, tenantService);
+
+        assertThatCode(loader::afterPropertiesSet).doesNotThrowAnyException();
+
+        String extensionSchema = plainJdbcTemplate.queryForObject(
+            "SELECT n.nspname FROM pg_extension e JOIN pg_namespace n ON e.extnamespace = n.oid "
+                + "WHERE e.extname = 'vector'",
+            String.class);
+
+        assertThat(extensionSchema).isEqualTo("public");
+
+        Integer schemaCount = plainJdbcTemplate.queryForObject(
+            "SELECT count(*) FROM information_schema.schemata WHERE schema_name = ?", Integer.class, VECTOR_SCHEMA);
+
+        assertThat(schemaCount).isEqualTo(1);
+
+        Integer tableCount = plainJdbcTemplate.queryForObject(
+            "SELECT count(*) FROM information_schema.tables WHERE table_schema = ? AND table_name = ?", Integer.class,
+            VECTOR_SCHEMA, TABLE_NAME);
+
+        assertThat(tableCount).isEqualTo(1);
+    }
+}

--- a/server/libs/modules/components/ai/llm/router/src/main/java/com/bytechef/component/ai/llm/router/model/RouterChatModel.java
+++ b/server/libs/modules/components/ai/llm/router/src/main/java/com/bytechef/component/ai/llm/router/model/RouterChatModel.java
@@ -21,6 +21,7 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.util.ArrayList;
 import java.util.Base64;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -321,7 +322,7 @@ public abstract class RouterChatModel implements org.springframework.ai.chat.mod
     }
 
     public List<String> getStop() {
-        return stop;
+        return Collections.unmodifiableList(stop);
     }
 
     public Integer getSeed() {
@@ -353,7 +354,7 @@ public abstract class RouterChatModel implements org.springframework.ai.chat.mod
     }
 
     public Map<String, Double> getLogitBias() {
-        return logitBias;
+        return Collections.unmodifiableMap(logitBias);
     }
 
     public Double getFrequencyPenalty() {

--- a/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
+++ b/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
@@ -35,7 +35,6 @@ import net.schmizz.sshj.SSHClient;
 import net.schmizz.sshj.sftp.SFTPClient;
 import net.schmizz.sshj.transport.verification.PromiscuousVerifier;
 import net.schmizz.sshj.xfer.InMemorySourceFile;
-import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.TestInstance;
 import org.springframework.beans.factory.annotation.Value;
@@ -252,6 +251,8 @@ public class BaseFtpActionIntTest {
      * exactly.
      */
     protected static final int PASSIVE_DATA_PORT = 30121;
+    private static final int FTP_CONTAINER_START_MAX_ATTEMPTS = 6;
+    private static final long FTP_CONTAINER_START_RETRY_DELAY_MILLIS = 2000;
     protected static final String TEST_CONTENT = "Hello from FTP Integration Test!";
     protected static final String TEST_FILE_BASE_NAME = "test-document";
     protected static final String TEST_FILE_BASE_EXTENSION = ".txt";
@@ -269,27 +270,14 @@ public class BaseFtpActionIntTest {
      * host port and container port to be identical: the server embeds the port number in the PASV response, so the FTP
      * client connects directly to that port on the host — dynamic port mapping would break this contract.
      */
-    @SuppressWarnings("deprecation")
-    protected FixedHostPortGenericContainer<?> ftpContainer;
-    protected GenericContainer<?> sftpContainer;
+    static FixedHostPortGenericContainer<?> ftpContainer;
+    static GenericContainer<?> sftpContainer;
 
     @BeforeAll
     void setUpContainers() throws Exception {
-        ftpContainer = new FixedHostPortGenericContainer<>("delfer/alpine-ftp-server:latest")
-            .withEnv("USERS", ftpUsername + "|" + ftpPassword)
-            .withEnv("ADDRESS", ftpHostIp)
-            .withEnv("MIN_PORT", String.valueOf(PASSIVE_DATA_PORT))
-            .withEnv("MAX_PORT", String.valueOf(PASSIVE_DATA_PORT))
-            .withExposedPorts(21)
-            .withFixedExposedPort(PASSIVE_DATA_PORT, PASSIVE_DATA_PORT);
+        startContainersOnce();
 
-        ftpContainer.start();
-
-        sftpContainer = new GenericContainer<>("atmoz/sftp:latest")
-            .withCommand(ftpUsername + ":" + ftpPassword + ":::" + SFTP_UPLOAD_DIR)
-            .withExposedPorts(22);
-
-        sftpContainer.start();
+        resetFtpFiles();
 
         uploadTestFileViaFtp();
         uploadTestFileViaSftp();
@@ -303,14 +291,58 @@ public class BaseFtpActionIntTest {
         copyTestFileViaFtp("005");
     }
 
-    @AfterAll
-    void tearDownContainers() {
-        if (ftpContainer != null) {
-            ftpContainer.stop();
+    private void startContainersOnce() throws InterruptedException {
+        if (ftpContainer == null) {
+            ftpContainer = startFtpContainer();
         }
 
-        if (sftpContainer != null) {
-            sftpContainer.stop();
+        if (sftpContainer == null) {
+            GenericContainer<?> container = new GenericContainer<>("atmoz/sftp:latest")
+                .withCommand(ftpUsername + ":" + ftpPassword + ":::" + SFTP_UPLOAD_DIR)
+                .withExposedPorts(22);
+
+            container.start();
+
+            sftpContainer = container;
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    private FixedHostPortGenericContainer<?> startFtpContainer() throws InterruptedException {
+        RuntimeException startException = null;
+
+        for (int attempt = 1; attempt <= FTP_CONTAINER_START_MAX_ATTEMPTS; attempt++) {
+            FixedHostPortGenericContainer<?> container =
+                new FixedHostPortGenericContainer<>("delfer/alpine-ftp-server:latest")
+                    .withEnv("USERS", ftpUsername + "|" + ftpPassword)
+                    .withEnv("ADDRESS", ftpHostIp)
+                    .withEnv("MIN_PORT", String.valueOf(PASSIVE_DATA_PORT))
+                    .withEnv("MAX_PORT", String.valueOf(PASSIVE_DATA_PORT))
+                    .withExposedPorts(21)
+                    .withFixedExposedPort(PASSIVE_DATA_PORT, PASSIVE_DATA_PORT);
+
+            try {
+                container.start();
+
+                return container;
+            } catch (RuntimeException exception) {
+                startException = exception;
+
+                container.stop();
+
+                Thread.sleep(FTP_CONTAINER_START_RETRY_DELAY_MILLIS);
+            }
+        }
+
+        throw new IllegalStateException(
+            "FTP container failed to start after " + FTP_CONTAINER_START_MAX_ATTEMPTS + " attempts", startException);
+    }
+
+    protected void resetFtpFiles() throws Exception {
+        var result = ftpContainer.execInContainer("sh", "-c", "rm -rf /ftp/" + ftpUsername + "/*");
+
+        if (result.getExitCode() != 0) {
+            throw new IllegalStateException("Failed to clean FTP test files inside container: " + result.getStderr());
         }
     }
 

--- a/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
+++ b/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
@@ -238,11 +238,11 @@ public class BaseFtpActionIntTest {
      * Shared credentials for both containers. Password must satisfy Alpine Linux's strength requirements used by
      * {@code delfer/alpine-ftp-server}'s entrypoint.
      */
-    @Value("${bytechef.test.ftp.password}")
+    @Value("${bytechef.test.ftp.password:T3st@Pass123}")
     protected String ftpPassword;
-    @Value("${bytechef.test.ftp.username}")
+    @Value("${bytechef.test.ftp.username:ftpuser}")
     protected String ftpUsername;
-    @Value("${bytechef.test.ftp.host.ip}")
+    @Value("${bytechef.test.ftp.host.ip:127.0.0.1}")
     protected String ftpHostIp;
 
     /**

--- a/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
+++ b/server/libs/modules/components/ftp/src/test/java/com/bytechef/component/ftp/action/BaseFtpActionIntTest.java
@@ -292,11 +292,19 @@ public class BaseFtpActionIntTest {
     }
 
     private void startContainersOnce() throws InterruptedException {
-        if (ftpContainer == null) {
+        if (ftpContainer == null || !ftpContainer.isRunning()) {
+            if (ftpContainer != null) {
+                ftpContainer.stop();
+            }
+
             ftpContainer = startFtpContainer();
         }
 
-        if (sftpContainer == null) {
+        if (sftpContainer == null || !sftpContainer.isRunning()) {
+            if (sftpContainer != null) {
+                sftpContainer.stop();
+            }
+
             GenericContainer<?> container = new GenericContainer<>("atmoz/sftp:latest")
                 .withCommand(ftpUsername + ":" + ftpPassword + ":::" + SFTP_UPLOAD_DIR)
                 .withExposedPorts(22);

